src/App.tsx

```diff
@@ -14,11 +15,94 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<unknown, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  removeGood = (good: string) => (
+    this.setState(prevState => ({
+      selectedGoods: prevState.selectedGoods.filter(myGood => myGood !== good),
+    }))
+  );
+
+  addGood = (good: string) => (
+    this.setState(prevState => ({
+      selectedGoods: [...prevState.selectedGoods, good],
+    }))
+  );
+
+  goodsCreator = (selectedGoods: string[]) => {
+    let goods = '';
+
+    if (selectedGoods.length === 1) {
+      goods = `is ${selectedGoods[0]}`;
+    } else if (selectedGoods.length === 2) {
+      goods = `are ${selectedGoods[0]} and ${selectedGoods[1]}`;
+    } else {
+      goods += 'are ';
+
+      for (let i = 0; i <= selectedGoods.length - 1; i += 1) {
+        if (i < selectedGoods.length - 2) {
+          goods += selectedGoods[i];
+          goods += ', ';
+        } else if (i === selectedGoods.length - 2) {
+          goods += selectedGoods[i];
+          goods += ' and ';
+        } else {
+          goods += selectedGoods[i];
+        }
+      }
+    }
+
+    return goods;
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <main className="main">
+          <h1 className="title">
+            {
+              selectedGoods.length
+                ? `Selected goods ${this.goodsCreator(selectedGoods)}`
+                : 'No goods selected'
+            }
+          </h1>
```

Comment:
Add button X for to the h1 to clear the selection.

Reply:
thank you, forgot about this button)